lib/remote-objects.js

```diff
@@ -160,12 +160,46 @@ RemoteObjects.prototype.invoke = function(method, ctorArgs, args, callback) {
 
 /**
  * Get an adapter by name.
- * @param {String} name The adapter name
+ * @param {String} nameOrClass The adapter name or provided class
  * @return {Adapter}
  */
 
-RemoteObjects.prototype.adapter = function(name) {
-  return require('./' + name + '-adapter');
+RemoteObjects.prototype.adapter = function(nameOrClass) {
+  if (typeof nameOrClass === 'function') {
+    // Throws an error if the provided adapter doesn't meets the requirements
+    this.validateAdapter(nameOrClass);
+
+    return nameOrClass;
+  } else {
+    return require('./' + nameOrClass + '-adapter');
+  }
+};
+
+RemoteObjects.prototype.getRequiredAdapterMethods = function() {
+  return ['createHandler'];
+};
+
+RemoteObjects.prototype.validateAdapter = function(Adapter) {
+  var requiredMethods = this.getRequiredAdapterMethods();
+
+  var notImplementedMethods = requiredMethods.reduce(
+      function(notImplementedMethods, methodString) {
+        if (typeof Adapter.prototype[methodString] !== 'function') {
+          return notImplementedMethods.concat(methodString);
+        }
+
+        return notImplementedMethods;
+      }, []);
+
+  if (notImplementedMethods.length > 0) {
+    throw new Error(g.f(
+        'Invalid adapter class. ' +
+        'The provided adapter class doesn\'t implement ' +
+        'the required methods: ' + notImplementedMethods.join(',')
```

Comment:
`g.f` behaves like `util.format`, with the difference that the first argument is translated before performing string replacements on `%s` and similar. The variable part of the message should be passed as the second argument.

AFAIK, there is a typographic convention to add a space after a comma, therefore the `join()` separator should be `', '`.

I am reading "the required methods"  as "all required methods", whereas the error shows only the methods that are missing. I think "the following required methods" may be a better formulation.

```js
throw new Error(g.f(
  'Invalid adapter class. ' +
    'The provided adapter class does not implement ' +
    'the following required methods: %s',
  notImplementedMethods.join(', ')));
```